### PR TITLE
Make an abstract contract that can be frozen. Does this by adding an …

### PR DESCRIPTION
**Pull request by edmundedgar**
…empty modifier to the parent contracts. Adds an example contract for testing.

### CODE REVIEW

File: packages/contracts/tests/python/test.py [hunk 1]
```diff
@@ -2639,7 +2644,7 @@ def test_too_soon_bonds_under_unrevealed_commit(self):
         
         self.assertEqual(self.rc0.functions.balanceOf(k3).call(), claimable)
 
-    #@unittest.skipIf(WORKING_ONLY, "Not under construction")
+    @unittest.skipIf(WORKING_ONLY, "Not under construction")
```
Comment (josojo) [inline packages/contracts/tests/python/test.py:2647]: is this still a todo?

File: packages/contracts/tests/python/test.py [hunk 2]
```diff
@@ -2708,6 +2713,106 @@ def test_is_history_of_unfinalized_question_valid(self):
...
+        txid = self.arb0.functions.requestArbitration(question_id, 0).transact(self._txargs(val=fee))
+        with self.assertRaises(TransactionFailed):
+            self.raiseOnZeroStatus(txid)
+
+        # The question will no longer finalize, even after the time has elapsed
+        self.assertFalse(self.rc0.functions.isFinalized(question_id).call())
+        self._advance_clock(40)
+        self.assertFalse(self.rc0.functions.isFinalized(question_id).call())
+        # Can't get result as not finalized
+        with self.assertRaises(TransactionFailed):
```
Comment (josojo) [inline packages/contracts/tests/python/test.py:2794]: it might be better to test for the exact error message that one should receive

**Review by josojo — APPROVED** [review]
Nice, that the test seems to pass. 

note: I would prefer if we would write out all variable names in the test, its hard to keep track of k1..k5, rc0,st...
If a good IDE, its zero more effort to write, but so much easier to read.

File: packages/contracts/development/contracts/RealityETHFreezableExample_ERC20-4.0.sol [hunk 3]
```diff
@@ -0,0 +1,17 @@
+// SPDX-License-Identifier: GPL-3.0-only
+
+pragma solidity ^0.8.20;
```
[the remark this answers is not in the document]
Comment (edmundedgar) [reply packages/contracts/development/contracts/RealityETHFreezableExample_ERC20-4.0.sol:3]: Might reorganize to break out testing stuff and interface stuff and so forth but if so I'll do it in another PR

**Pull request closed without merging**